### ISSUE TITLE
feature request: add detection of INTEL-SA-00075

### ISSUE THREAD

**Issue by skochinsky** [chipsec/chipsec#212]
reference: https://security-center.intel.com/advisory.aspx?intelid=INTEL-SA-00075&languageid=en-fr

This consists of two parts:
1. determine ME version
2. detect if AMT is supported and enabled/provisioned

1 can be done using using HECI commands. for HECI communication, either the OS driver can be used, or HECI protocol can be reimplemented over PCI to the ME device (like in `intelmetool`). Then the MKHI command `GEN_GET_MKHI_VERSION` can be used to retrieve the firmware version. I have an example [here](https://github.com/skochinsky/me-tools/blob/master/me_util.py) (uses Windows HECI driver).

2 can be done using `MKHI_FWCAPS_GET_RULE` command (see [` mkhi_get_fwcaps`  in `intelmetool`](https://github.com/zamaudio/intelmetool/blob/main/me.c#L428) )

**Comment by ismaws** [comment]
tried intelmetool on a couple of xeon-based arch, seems it does not fully support servers. (keeps returning "Hardware unsupported by intelmetool, exiting")
 
any recommended recipe for Xeon/Server detection?

**Comment by skochinsky** [comment]
@ismaws well, the official tool is SpsInfo (available in SPS firmware kits from Intel) but I don't know any open-source implementations. you can try adding your device ID to the macros in `intelmetool.h` and see what happens. Anyway, it's probably better discussed over there.

**Comment by c7zero** [comment]
Yup, makes sense to add a check. It's better to implement HECI to ME in CHIPSEC, in a separate HAL component. Thanks! Will look into it.

**Comment by npmitche** [comment]
New active link: https://www.intel.com/content/www/us/en/security-center/advisory/intel-sa-00075.html

**Comment by dscott90** [comment]
https://github.com/coreboot/coreboot/tree/master/util/intelmetool

Potential tool for exposing intel ME version.

Possible solution: Create new chipsec module that exposes intel ME version based on _intelmetool_ functionality -> Use ME version to determine if SUT is vulnerable based on the advisory commented above.

**Comment by raplasce** [comment]
@npmitche @dscott90 

I think I need a little help getting started on this one.

1. Where's the best place to store this new module?
2. I haven't been able to use the Intelmetool successfuly, I've built the intelmetool from source but just get this:
    <img width="945" height="269" alt="Image" src="https://github.com/user-attachments/assets/f47fce3e-e8c5-4d72-8852-f84e81cb7db9" />
3. I've at least been able to follow the instructions given the security advisory on my computer, but I don't think it helps me much:
    <img width="1019" height="588" alt="Image" src="https://github.com/user-attachments/assets/14edc426-6c50-48da-a894-067addcc8ff4" />

**Comment by dscott90** [comment]
I believe this new module would fit best in the 'chipsec/modules/common' folder. A name like imf_eop.py (Intel Manageability Firmware Escalation of Privilege) would be descriptive enough.

The key piece of information we need to gather is the Intel ME version. I don't believe the CMSE version detection tool exposes the functionality used. My thinking is that the coreboot intelmetool functionality for Intel ME version retrieval can be extracted for use in the new chipsec module.
